src/main/java/org/sonar/plugins/csharp/RoslynProfileExporter.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * SonarQube C# Plugin
+ * Copyright (C) 2014-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.csharp;
+
+import com.google.common.base.Throwables;
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ListMultimap;
+import com.google.common.collect.Sets;
+import org.apache.commons.codec.binary.Base64;
+import org.sonar.api.config.PropertyDefinition;
+import org.sonar.api.config.Settings;
+import org.sonar.api.profiles.ProfileExporter;
+import org.sonar.api.profiles.RulesProfile;
+import org.sonar.api.rules.ActiveRule;
+import org.sonar.api.server.rule.RulesDefinition;
+import org.sonar.api.server.rule.RulesDefinition.Context;
+import org.sonar.api.server.rule.RulesDefinition.Repository;
+import org.sonar.api.server.rule.RulesDefinition.Rule;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+
+public class RoslynProfileExporter extends ProfileExporter {
+
+  private static final String SONARLINT_PLUGIN_KEY = "sonarlint-cs";
+  private static final String ROSLYN_REPOSITORY_PREFIX = "roslyn-";
+  private final Settings settings;
+  private final RulesDefinition[] rulesDefinitions;
+
+  public RoslynProfileExporter(Settings settings, RulesDefinition[] rulesDefinitions) {
+    super("roslyn-cs", "Technical exporter for the MSBuild SonarQube Scanner");
+    this.settings = settings;
+    this.rulesDefinitions = rulesDefinitions;
+    setSupportedLanguages(CSharpPlugin.LANGUAGE_KEY);
+  }
+
+  // TODO Validation of required plugin properties
+  // TODO XML escaping
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1135)